### PR TITLE
[feaLib] show all missing glyphs at once

### PR DESCRIPTION
**Pull request by simoncozens**
I just had a frustrating situation where the FEA parser reported a missing glyph in my source, and then I fixed it and re-ran it, and then it reported another missing glyph, and I fixed it and re-ran it, and...

With this PR, "missing glyph" errors are gathered and reported at the end of parsing:

```
ERROR: The following glyph names are referenced but are missing from the glyph set:
 ampersand (first found at source/build/features.fea:43570:16)
 asterisk (first found at source/build/features.fea:43605:14)
```

### CODE REVIEW

File: Lib/fontTools/feaLib/parser.py [hunk 1]
```diff
@@ -2079,13 +2089,11 @@ def check_glyph_name_in_glyph_set(self, *names):
...
-                raise FeatureLibError(
-                    "The following glyph names are referenced but are missing from the "
-                    f"glyph set: {', '.join(missing)}",
-                    self.cur_token_location_,
-                )
+            for name in names:
+                if name in self.glyphNames_:
+                    continue
+                if name not in self.missing:
+                    self.missing[name] = self.cur_token_location_
```
Comment (anthrotype) [inline Lib/fontTools/feaLib/parser.py:2096]: the docstring of `check_glyph_name_in_glyph_set` says "Raises if..." but it no longer raises, does it?

**Review by anthrotype — COMMENTED** [review]
I can see how that can be handy, however I wonder if it's a problem delaying the exception at the very end of the parsing and whether there's some code in between that may assume that given glyph names exist after running `check_glyph_name_in_glyph_set`.
But maybe it's fine.

**Comment by simoncozens** [comment]
You can create an ast.Glyph or ast.GlyphClass object with non-existent glyphs; it'll only fail when you try and build it. It's a semantic problem rather than a syntactic one. So reporting at the end of parsing is fine.

**Comment by anthrotype** [comment]
so technically the parser _could_ finish its job without raising then..

**Comment by anthrotype** [comment]
maybe Parser.parse() method could have an ignoreMissingGlyphs=False parameter.. well, nevermind. LGTM

**Review by anthrotype — APPROVED** [review]

**Pull request merged**